Integrations/integration-CIRCL.yml

```diff
@@ -0,0 +1,324 @@
+            'Contents': data,
+            'HumanReadable': '```json\n' + json.dumps(records, indent=4) + '\n```',
+            'EntryContext': {
+                'CIRCLssl.IP(val.Value===obj.Value)': records
+            }
+        }
+
+    def createIpContext(ip, ipData):
+        certificates = []
+
+        for sha1 in ipData['certificates']:
+            subjects = ipData['subjects'][sha1]['values']
+            certificates.append(createCertificateContext(sha1, subjects))
+
+        return {
+            'Value': ip,
+            'Certificate': certificates
+        }
+
+    def createCertificateContext(sha1, subjects):
+        return {
+            'Sha1': sha1,
+            'Subjects': subjects
+        }
+
+    def listCertificateSeenIps(sha1, limit):
+        queryUrl = BASE_URL + '/v2pssl/cquery/' + sha1
+
+        responce = requests.get(queryUrl, auth=AUTH)
+        responce.raise_for_status()
+
+        data = responce.json()
+        certificate = createCertificateSeenIpsContext(sha1, data, limit)
+
+        return {
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': data,
+            'HumanReadable': 'Hits: ' + str(certificate['Hits']),
+            'EntryContext': {
+                'CIRCLssl.Certificate(val.Sha1===obj.Sha1)': certificate,
+            }
+        }
+
+    def createCertificateSeenIpsContext(sha1, data, limit):
+        return {
+            'Sha1': sha1,
+            'Hits': data['hits'],
+            'IP': data['seen'][:limit],
+        }
+
+    def getCertificateDetails(sha1):
+        queryUrl = BASE_URL + '/v2pssl/cfetch/' + sha1
+
+        responce = requests.get(queryUrl, auth=AUTH)
+        responce.raise_for_status()
+
+        data = responce.json()
+        certificate = createCertificateDetails(sha1, data)
+
+        return {
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': data,
+            'HumanReadable': certificate,
+            'EntryContext': {
+                'CIRCLssl.Certificate(val.Sha1===obj.Sha1)': certificate,
+            }
+        }
+
+    def createCertificateDetails(sha1, data):
+        info = data['info']
+        extension = info['extension']
+
+        return {
+            'Sha1': sha1,
+            'Usage': extension['extendedKeyUsage'],
+            'Distribution': extension['crlDistributionPoints'],
+            'Issuer': info['issuer'],
+            'Time': info['not_before'],
+            'Subject': info['subject'],
+            'Key': info['key'],
+            'Pem': data['pem'],
+            'Seen': data.get('icsi', {}).get('times_seen'),
+        }
+
+    try:
+        args = demisto.args()
+
+        if demisto.command() == 'test-module':
+            responce = dnsGet('test.com')
+            demisto.results('ok')
+
+        if demisto.command() == 'circl-dns-get':
+            demisto.results(dnsGet(args.get('queryValue')))
```

Comment:
Please use `demisto.results()` in the functions, and here only call the functions

Reply:
Thats what I saw from other integrations.
I think its nice to keep the functions pure from demisto commands (when possible).
Any way I'll change it a little and tell me what you think.

Reply:
I think its better now. Im resolving this one because its outdated. Please take a look at the new code.